### ISSUE TITLE
"undefined method `[]' for nil:NilClass" when running "rake middleware" task

### ISSUE THREAD

**Issue by knightq** [rsim/oracle-enhanced#17]
# 

```
  M Y    C O N F I G U R A T I O N
```
# 

RUBY:   ruby-1.8.7-p249 (on rvm)
RAILS:  Rails 3.0.0.beta4
GEMS: 
    **\* LOCAL GEMS ***
    abstract (1.0.0)
    actionmailer (3.0.0.beta4, 3.0.0.beta)
    actionpack (3.0.0.beta4, 3.0.0.beta)
    activemodel (3.0.0.beta4, 3.0.0.beta)
    activerecord (3.0.0.beta4, 3.0.0.beta)
    activerecord-oracle_enhanced-adapter (1.2.4)
    activeresource (3.0.0.beta4, 3.0.0.beta)
    activesupport (3.0.0.beta4, 3.0.0.beta)
    arel (0.4.0, 0.2.1)
    builder (2.1.2)
    bundler (1.0.0.beta.1, 0.9.26)
    cgi_multipart_eof_fix (2.5.0)
    daemons (1.1.0, 1.0.10)
    erubis (2.6.6, 2.6.5)
    fastthread (1.0.7)
    gem_plugin (0.2.3)
    i18n (0.4.1, 0.3.7)
    jrails (0.6.0)
    mail (2.2.5, 2.2.3, 2.1.5.3)
    memcache-client (1.8.3, 1.7.8)
    mime-types (1.16)
    mongrel (1.2.0.pre2, 1.1.5)
    pdfkit (0.3.3)
    polyglot (0.3.1)
    rack (1.1.0)
    rack-mount (0.6.4, 0.6.3, 0.4.7)
    rack-test (0.5.4)
    rails (3.0.0.beta4, 3.0.0.beta)
    railties (3.0.0.beta4, 3.0.0.beta)
    rake (0.8.7)
    rdoc (2.5.8)
    rspec (1.3.0)
    ruby-graphviz (0.9.12)
    ruby-oci8 (2.0.4)
    ruby-plsql (0.4.3)
    sqlite3-ruby (1.3.0)
    text-format (1.0.0)
    text-hyphen (1.0.0)
    thor (0.13.7, 0.13.6)
    treetop (1.4.8)
    tzinfo (0.3.22)

GEMS included by the bundle:

```
abstract (1.0.0)
actionmailer (3.0.0.beta4)
actionpack (3.0.0.beta4)
activemodel (3.0.0.beta4)
activerecord (3.0.0.beta4)
activerecord-oracle_enhanced-adapter (1.3.0 6f204a4)
activeresource (3.0.0.beta4)
activesupport (3.0.0.beta4)
arel (0.4.0)
authlogic (2.1.3 a087ad0)
builder (2.1.2)
bundler (1.0.0.beta.1)
erubis (2.6.5)
i18n (0.4.1)
jrails (0.6.0)
mail (2.2.5)
mime-types (1.16)
pdfkit (0.3.3)
polyglot (0.3.1)
rack (1.1.0)
rack-mount (0.6.4)
rack-test (0.5.4)
rails (3.0.0.beta4 0421fb7)
railties (3.0.0.beta4)
rake (0.8.7)
rspec (1.3.0)
ruby-oci8 (2.0.4)
ruby-plsql (0.4.3)
thor (0.13.6)
treetop (1.4.8)
tzinfo (0.3.22)
```
# 

When I try to run 'rake middleware' I got an error "undefined method `[]' for nil:NilClass".
Using the same command with the option '--trace', as suggested by the output, I see that it seems a problem of oracle_enhanced_tasks.rb, here is the full output of the command 'rake middleware --trace':

```
rake aborted!
undefined method `[]' for nil:NilClass
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/bundler/gems/oracle-enhanced-6f204a4/lib/active_record/connection_adapters/oracle_enhanced_tasks.rb:3
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/bundler/gems/rails-0421fb7/activesupport/lib/active_support/dependencies.rb:212:in `require'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/bundler/gems/rails-0421fb7/activesupport/lib/active_support/dependencies.rb:212:in `require'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/bundler/gems/rails-0421fb7/activesupport/lib/active_support/dependencies.rb:198:in `load_dependency'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/bundler/gems/rails-0421fb7/activesupport/lib/active_support/dependencies.rb:554:in `new_constants_in'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/bundler/gems/rails-0421fb7/activesupport/lib/active_support/dependencies.rb:198:in `load_dependency'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/bundler/gems/rails-0421fb7/activesupport/lib/active_support/dependencies.rb:212:in `require'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/bundler/gems/oracle-enhanced-6f204a4/lib/active_record/connection_adapters/oracle_enhanced_adapter.rb:1777
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/gems/bundler-1.0.0.beta.1/lib/bundler/runtime.rb:48:in `require'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/gems/bundler-1.0.0.beta.1/lib/bundler/runtime.rb:48:in `require'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/gems/bundler-1.0.0.beta.1/lib/bundler/runtime.rb:47:in `each'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/gems/bundler-1.0.0.beta.1/lib/bundler/runtime.rb:47:in `require'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/gems/bundler-1.0.0.beta.1/lib/bundler/runtime.rb:38:in `each'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/gems/bundler-1.0.0.beta.1/lib/bundler/runtime.rb:38:in `require'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/gems/bundler-1.0.0.beta.1/lib/bundler.rb:112:in `require'
/home/asalicetti/Ruby/Rails/GS3/config/application.rb:7
/home/asalicetti/.rvm/rubies/ruby-1.8.7-p249/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in `gem_original_require'
/home/asalicetti/.rvm/rubies/ruby-1.8.7-p249/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in `require'
/home/asalicetti/Ruby/Rails/GS3/Rakefile:4
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/gems/rake-0.8.7/lib/rake.rb:2383:in `load'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/gems/rake-0.8.7/lib/rake.rb:2383:in `raw_load_rakefile'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/gems/rake-0.8.7/lib/rake.rb:2017:in `load_rakefile'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/gems/rake-0.8.7/lib/rake.rb:2068:in `standard_exception_handling'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/gems/rake-0.8.7/lib/rake.rb:2016:in `load_rakefile'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/gems/rake-0.8.7/lib/rake.rb:2000:in `run'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/gems/rake-0.8.7/lib/rake.rb:2068:in `standard_exception_handling'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/gems/rake-0.8.7/lib/rake.rb:1998:in `run'
/home/asalicetti/.rvm/gems/ruby-1.8.7-p249/gems/rake-0.8.7/bin/rake:31
/home/asalicetti/.rvm/rubies/ruby-1.8.7-p249/bin/rake:24:in `load'
/home/asalicetti/.rvm/rubies/ruby-1.8.7-p249/bin/rake:24
```

I've done a little debug and it turns out that 'Rails.env', at runtime, is equal to 'development', but Rails.env['adapter'] is undefined.

Looking on Google I found this: 
http://groups.google.com/group/oracle-enhanced/browse_thread/thread/7e246a9b85c4b2ed/a48a06f8946bd400

Basically the solution suggested here is to remove the bit of code:
    , :require => 'active_record/connection_adapters/oracle_enhanced_adapter'
from the line of the activerecord-oracle_enhanced-adapter gem's requirement.

It works!

My question is: does it have any drawback such kind of fix?
I don't remember even why I put it on there...

Thank you!
